app/models/katello/content_view.rb

```diff
@@ -770,43 +770,30 @@ def related_composite_cvs
       content_views
     end
 
-    def audited_cv_repositories_since_last_publish
-      audited_repositories = self.audits.filter do |a|
+    def audited_cv_repositories_since_last_publish?
+      self.audits.where("created_at > ?", latest_version_object.created_at).any? do |a|
         !a.audited_changes["repository_ids"].nil?
       end
```

Comment:
The `.any?` here uses Ruby (which is slow) to do the bulk of the work. I see that `audited_changes` is a `text` field that holds JSON, which Rails/Ruby converts into a Hash for us. I wonder if there's a way to use SQL to accomplish the equivalent of `!a.audited_changes["repository_ids"].nil?` If we could incorporate that into our `where` somehow, Postgres (which is much faster than Ruby) could do all these queries.

Reply:
Pushed a commit with this. This is what I have with ~70k audit records:
![Screenshot from 2024-02-29 14-43-55](https://github.com/Katello/katello/assets/21146741/af10cfd8-497a-46fe-904d-15bc71fad375)